### DISCUSSION TITLE
It should be possible to press the OK button via a keyboard shortcut

### DISCUSSION THREAD

**Discussion by RobustoSystems** [ni/labview-icon-editor#206] [General]
**Problem**: Currently pressing the OK button requires using the mouse.

**Solution**: It would be useful if pressing the OK button could be done via a keyboard shortcut. For example, Ctrl + Enter could be equivalent to pressing OK. Ctrl + Enter is already used as a shortcut for the "proceed" actions in some applications. For example, Ctrl + Enter is equivalent to pressing the "Send" button when creating an email in Outlook, and is equivalent to pressing the "Commit" or "Commit & Push" button when making a commit using TortoiseGit. Ctrl + Enter is just an example - I would be happy with any other key combination.

Thanks!

**Comment by svelderrainruiz** [comment]
@RobustoSystems  this is not a big change, and makes sense to me. I am moving this to phase 2 since we are rearchitecting the icon editor first before creating new dev work.

would you be interested on working on this issue? after submitting your pull request i can create a new release with this fix, and you would have a package you can install made by the CI/CD pipeline i just created.

also please let me know if you are willing on joining the icon editor technical steering team, you make a good fit.

**Comment by RobustoSystems** [comment]
Hi Sergio,

Thanks, that all makes sense.

_"would you be interested on working on this issue?"_ - Yes, I would be interested in implementing this feature. Should I wait until the rearchitecting is complete or is the codebase ready for this change?

_"also please let me know if you are willing on joining the icon editor technical steering team, you make a good fit."_ - Thanks! Yes, I would be honored to be part of the icon editor technical steering team.

Kind regards,
Petru Tarabuta, Robusto Systems Ltd, UK

**Comment by svelderrainruiz** [comment]
> Yes, I would be interested in implementing this feature. Should I wait until the rearchitecting is complete or is the codebase ready for this change?

No need to wait for the rearchitecting, the code base is ready and we had our first contribution already. I would appreciate if you have any issues so that i can elaborate more on the documentation.

> Thanks! Yes, I would be honored to be part of the icon editor technical steering team.

Perfect, just sent you a DM through linkedin. Welcome to the team :)

I am assigning this work to you, see you on thursday!

**Comment by RobustoSystems** [comment]
After a quick investigation it seems that it will not be possible to implement this functionality (the "Ctrl + Enter" keyboard shortcut) by simply modifying the Key Navigation properties of the OK button. This is shown in the screenshot below.

![Image](https://github.com/user-attachments/assets/1bf3c3ab-c718-4c0e-be7e-0b99e25fb3d4)

The functionality could be implemented by intelligently handling the "Key Down" event inside an Event Structure of the VI.

**Comment by svelderrainruiz** [comment]
We are going to change it to a discussion based on conversations with @crossrulz , @j-medland  and Fab

**Comment by crossrulz** [comment]
Interestingly, the OK button is already set to receive focus on the return key. So if nothing else is being done (ex: text entry), hitting the return key twice will act as a toggle (once to set focus, a second to activate the button).

Is this good enough to satisfy the request?
